resource.go

```diff
@@ -93,3 +100,36 @@ func New(config Config) (*Resource, error) {
 func (r *Resource) Name() string {
 	return Name
 }
+
+func (r *Resource) applyPatches(ctx context.Context, accessor metav1.Object, patches []Patch) error {
+	patches = append(patches, Patch{
+		Op:    "test",
+		Value: accessor.GetResourceVersion(),
+		Path:  "/metadata/resourceVersion",
+	})
+
+	b, err := json.Marshal(patches)
+	if err != nil {
+		return microerror.Mask(err)
+	}
+	p := ensureSelfLink(accessor.GetSelfLink())
+
+	err = r.restClient.Patch(types.JSONPatchType).AbsPath(p).Body(b).Do().Error()
+	if errors.IsConflict(err) {
+		return microerror.Mask(err)
+	} else if errors.IsResourceExpired(err) {
+		return microerror.Mask(err)
+	} else if err != nil {
+		return microerror.Mask(err)
+	}
```

Comment:
Note the weirdness here with the error handling. This is for debugging and find out what kind of error we get when the patch fails due to the failed resource version test. Never happened so far. Just keeping for a while.

Reply:
Cool thanks for the explanation.

Reply:
I would leave it as it is even after debugging. It makes it clear what errors this function can produce.